test/service/test_manifest.py

```diff
@@ -204,6 +207,41 @@ def test_pfb_manifest(self):
                         with open(results_file, 'w') as f:
                             json.dump(records, f, indent=4, sort_keys=True)
 
+    def shared_file_bundle(self, bundle):
+        bundle = self._load_canned_bundle(bundle)
+
+        process = bundle.metadata_files['process_0.json']
+        process_id = '17354879-5237-4f67-9954-7cdab2ee8d0a'
+        process['provenance']['document_id'] = process_id
+        bundle.manifest[1]['uuid'] = process_id
+
+        cell_suspension = bundle.metadata_files['cell_suspension_0.json']
+        cell_suspension_id = 'd6b3d2ab-5715-4486-a544-ac09fafac279'
+        cell_suspension['provenance']['document_id'] = cell_suspension_id
+        bundle.manifest[4]['uuid'] = cell_suspension_id
+
+        specimen = bundle.metadata_files['specimen_from_organism_0.json']
+        specimen_id = '5275e5a0-6043-4ec9-86a1-6c1140cbeede'
+        specimen['provenance']['document_id'] = specimen_id
+        bundle.manifest[9]['uuid'] = specimen_id
+        # Change organ to prevent cell_suspensions from aggregating together
+        specimen['organ'] = {
+            "text": "lung",
+            "ontology": "UBERON:0002048",
+            "ontology_label": "lung"
+        }
+
+        links = bundle.metadata_files['links.json']
+        one(links['links'][0]['inputs'])['input_id'] = cell_suspension_id
+        one(links['links'][1]['outputs'])['output_id'] = cell_suspension_id
+        one(links['links'][1]['inputs'])['input_id'] = specimen_id
+        one(links['links'][2]['outputs'])['output_id'] = specimen_id
+        links['links'][2]['process_id'] = process_id
```

Comment:
This is all a bit arcane. Consider a generic, recursive search and replace.

Reply:
Using string replace instead.